### PR TITLE
[WIP] Synchronize users removed from teams

### PR DESCRIPTION
**Pull request by Kadaaran**
Hi folks ! 👋 

First fix for the issue #1216 / #1215 
TODO:
- [ ] Write tests
- [ ] Removed from organization as well ?

It seems Classroom does not accept some events, especially when a user is removed from an organization or leaves a team in GitHub.
The thing is that in Classroom the number of members was wrong all this time (the informations were not updated at all).

I found a way to fix it. Not sure if it's a good way to do it but it works.

Next step would be to do the same when someone leaves an organization or is removed by an admin. 
I realized after I've finished it that someone was already working on the organization/teams issue #1174 🙈 

screen shot 2018-01-25 at 15 13 51
screen shot 2018-01-25 at 18 10 57

Happy to hear your comments about it ✨

### CODE REVIEW

File: app/jobs/membership_event_job.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  def perform(payload_body)
+    return true unless payload_body.dig("action") == "removed"
+
+    user_id = payload_body.dig("member", "id")
+    team_id = payload_body.dig("team", "id")
+    user = User.find_by(uid: user_id)
+    group_id = Group.find_by(github_team_id: team_id).id
+
+    repo_accesses = GroupAssignmentRepo.find_by(group_id: group_id).repo_accesses
```
Comment (tarebyte) [inline app/jobs/membership_event_job.rb]: So `Group` has `repo_accesses` directly https://github.com/education/classroom/blob/44efed0852076877840d2dcd387389f165e0eaa5/app/models/group.rb#L13-L14

So you could just do:

```ruby
def perform(payload_body)
  return true unless payload_body.dig("action") == "removed"
  
  user_id = payload_body.dig("member", "id")
  team_id = payload_body.dig("team", "id")

  user  = User.find_by(uid: user_id)
  return true unless user.present?

  group = Group.find_by(github_team_id: team_id)
  return true unless group.present?

  group.repo_accesses.find_by(user_id: user.id)
  repo_access&.destroy
end
```

Comment (Kadaaran) [reply app/jobs/membership_event_job.rb]: I get it now thanks ! 👍

**Comment by Kadaaran** [comment]
Duplicate of #1174

**Comment by Kadaaran** [comment]
Duplicate of #1174

**Pull request closed without merging**